src/Analyser/TypeSpecifier.php

```diff
@@ -1096,87 +1101,6 @@ private function specifyTypesForConstantBinaryExpression(
 			));
 		}
 
-		if (
```

Comment:
moved this 2 if-blocks out of `specifyTypesForConstantBinaryExpression` as we no longer check for a constant type operand, but do a more general `isInteger` based handling, which includes unbound `IntegerRangeType`.